diesel/src/pg/query_builder/tablesample.rs

```diff
@@ -0,0 +1,213 @@
+use crate::expression::{Expression, ValidGrouping};
+use crate::pg::Pg;
+use crate::query_builder::{AsQuery, AstPass, FromClause, QueryFragment, QueryId, SelectStatement};
+use crate::query_source::QuerySource;
+use crate::result::QueryResult;
+use crate::sql_types::{Double, SmallInt};
+use crate::{JoinTo, SelectableExpression, Table};
+use std::marker::PhantomData;
+
+#[doc(hidden)]
+pub trait TablesampleMethod: Clone {
+    fn method_name_sql() -> &'static str;
+}
+
+#[derive(Clone, Copy, Debug)]
+/// Used to specify the `BERNOULLI` sampling method.
+pub struct BernoulliMethod;
+
+impl TablesampleMethod for BernoulliMethod {
+    fn method_name_sql() -> &'static str {
+        "BERNOULLI"
+    }
+}
+
+#[derive(Clone, Copy, Debug)]
+/// Used to specify the `SYSTEM` sampling method.
+pub struct SystemMethod;
+
+impl TablesampleMethod for SystemMethod {
+    fn method_name_sql() -> &'static str {
+        "SYSTEM"
+    }
+}
+
+/// Represents a query with a `TABLESAMPLE` clause.
+#[doc(hidden)]
+#[derive(Debug, Clone, Copy)]
+pub struct Tablesample<S, TSM>
+where
+    TSM: TablesampleMethod,
+{
+    pub source: S,
+    method: PhantomData<TSM>,
+    portion: i16,
+    seed: Option<f64>,
+}
+
+impl<S, TSM> Tablesample<S, TSM>
+where
+    TSM: TablesampleMethod,
+{
+    pub fn new(source: S, portion: i16) -> Tablesample<S, TSM> {
```

Comment:
```suggestion
    pub(crate) fn new(source: S, portion: i16) -> Tablesample<S, TSM> {
```

There is no need for someone outside of diesel to call this function, as query dsl structs are generally constructed via the corresponding trait methods.